src/controllers/sync-controller.ts

```diff
@@ -338,9 +387,12 @@ export class SyncController implements ReactiveController {
   /**
    * Register periodic background sync if capabilities are available and settings allow
    * Phase 1.3: App initialization periodic sync registration
+   * Note: Service worker handles duplicate registrations gracefully
    */
   async #registerPeriodicBackgroundSyncIfAvailable(): Promise<void> {
-    // Avoid duplicate registrations
+    // Track registration state to avoid unnecessary messages
+    // Note: The service worker will handle duplicate registrations gracefully,
+    // but we track state here to reduce unnecessary message passing
     if (this.#periodicSyncRegistered) {
       return;
     }
```

Comment:
I understand the desire to reduce unnecessary messages, but now you have to keep `this.#periodicSyncRegistered` in sync with the state inside the service worker. It's going to be less error-prone to just send the messages unconditionally so that we don't need to have separate state here.